tests/memory/enc/fragment.c

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/globals.h>
+#include <openenclave/internal/tests.h>
+
+#include <malloc.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "memory_t.h"
+
+#define MALLOC_SIZE_SMALL 1024
+#define EFFICIENCY_TEST_TIMES 5000000
+
+#define MALLINFO_FIELD_TYPE size_t
+struct mallinfo
+{
+    MALLINFO_FIELD_TYPE arena;    /* non-mmapped space allocated from system */
+    MALLINFO_FIELD_TYPE ordblks;  /* number of free chunks */
+    MALLINFO_FIELD_TYPE smblks;   /* always 0 */
+    MALLINFO_FIELD_TYPE hblks;    /* always 0 */
+    MALLINFO_FIELD_TYPE hblkhd;   /* space in mmapped regions */
+    MALLINFO_FIELD_TYPE usmblks;  /* maximum total allocated space */
+    MALLINFO_FIELD_TYPE fsmblks;  /* always 0 */
+    MALLINFO_FIELD_TYPE uordblks; /* total allocated space */
+    MALLINFO_FIELD_TYPE fordblks; /* total free space */
+    MALLINFO_FIELD_TYPE keepcost; /* releasable (via malloc_trim) space */
+};
+
+struct mallinfo dlmallinfo(void);
+
+static size_t _get_heap_size()
+{
+    // call malloc to update the globle data.
+    free(malloc(1));
+    return dlmallinfo().keepcost;
+}
+
+static int _malloc_free_fixed_size(size_t size, int times)
+{
+    int i = 0;
+    int* buffer = NULL;
+
+    for (i = 0; i < times; i++)
+    {
+        buffer = (int*)malloc(size);
+        if (NULL == buffer)
+        {
+            return -1;
+        }
+        free(buffer);
+    }
+
+    oe_host_printf("_malloc_free_fixed_size malloc times = %d.\n", i);
+    return 0;
+}
+
+static inline size_t _randx(size_t x)
+{
+    /* Note that rand() % N is biased if RAND_MAX + 1 isn't divisible
+     * by N. But, slight probability bias doesn't really matter in these
+     * tests. */
+    return (size_t)rand() % x;
+}
+
+static int _malloc_free_random_size(int times)
+{
+    int i = 0;
+    size_t size = 0;
+    void* buffer = NULL;
+
+    for (i = 0; i < times; i++)
+    {
+        size = _randx(_get_heap_size());
+        buffer = malloc(size);
+        if (NULL == buffer)
+        {
+            return -1;
+        }
+
+        free(buffer);
+    }
+    oe_host_printf("_malloc_free_random_size malloc times = %d.\n", i);
+    return 0;
+}
+
+void test_malloc_fixed_size_fragment(void)
+{
+    // get heap size before tests
+    size_t heap_size_before_test = _get_heap_size();
+    oe_host_printf(
+        "[test_malloc_fixed_size_fragment]heap size before test : %zu.\n",
+        heap_size_before_test);
+
+    OE_TEST(
+        _malloc_free_fixed_size(MALLOC_SIZE_SMALL, EFFICIENCY_TEST_TIMES) == 0);
+
+    size_t heap_size_after_test = _get_heap_size();
+    oe_host_printf(
```

Comment:
Is there a check that can be made to the heap size after as well?

Reply:
Good suggestion! Thanks